### ISSUE TITLE
Please remove l.facebook.com redirect

### ISSUE THREAD

**Issue by andreicristianpetcu** [idlewan/link_cleaner#1]
Great WebExtension! :)

Can you please remove Facebook useless redirects?

I have this link in Facebook

https://l.facebook.com/l.php?u=http%3A%2F%2Feconomie.hotnews.ro%2Fstiri-telecom-21702036-adio-taxe-roaming-15-iunie-2017-europenii-nu-vor-mai-avea-costuri-suplimentare-telefonie-internet-mobil-cand-vor-calatori-alt-stat.htm&h=ATP1kf98S0FxqErjoW8VmdSllIp4veuH2_m1jl69sEEeLzUXbkNXrVnzRMp65r5vf21LJGTgJwR2b66m97zYJoXx951n-pr4ruS1osMvT2c9ITsplpPU37RlSqJsSgba&s=1

I want it to redirect to this URL bypassing Facebook's tracking system:

http://economie.hotnews.ro/stiri-telecom-21702036-adio-taxe-roaming-15-iunie-2017-europenii-nu-vor-mai-avea-costuri-suplimentare-telefonie-internet-mobil-cand-vor-calatori-alt-stat.htm

So basically apply this regex take the first capturing group, decode it as base64 and then redirect to it:

https://l.facebook.com/l.php[?]u=(.*)&h=.*

I think if an addon does this it would be faster since sometimes l.facebook.com seems to hang on slow Internet connections.

Thank you!

**Comment by idlewan** [comment]
That's a good idea, I don't use facebook so I don't encounter these kinds of links.
The code is quite small and should be readable, do you want to try to add it in?
Look at these lines https://github.com/idlewan/link_cleaner/blob/master/background.js#L126-L141 on how to add a new listener for a specific site.

**Comment by andreicristianpetcu** [comment]
I will try :)

**Comment by idlewan** [comment]
I'm aware that it doesn't (a comment on addons.mozilla.org mentionned it), but I haven't taken the time yet to look at what the fix for nightly would be.

**Comment by andreicristianpetcu** [comment]
@idlewan Apparently it was missing the ["<all_urls>" permission](https://developer.mozilla.org/en-US/Add-ons/WebExtensions/manifest.json/permissions#activeTab_permission). It works now in my PR :)

**Comment by idlewan** [comment]
I might have used `URL` and `URLSearchParams` objects instead of a regex had I done it myself (might be more resilient if the order of search params ever change, although it's probably unlikely. I don't know the performance difference between these and a regex, might be worth a benchmark), but your version works fine, so it's all right by me.

Thanks very much for the PR, I pulled the changes and submitted a new version for review.
It should update in a few days, the reviewers have been quite fast for previous versions.

Good work!

**Comment by andreicristianpetcu** [comment]
I plan to add other junk redirect cleaning PRs if you wish. I can Implement it with URLSearchParams and extract the code into a common function. I don't care about client side performance. I want my browser not to hit other servers just to bounce off them with a redirect. When I click a link I want ti to go directly to that link.

**Comment by idlewan** [comment]
PRs of this kind will be welcome.

You can write it as you see fit, as long as it's readable and lean. If abstracting in a common function feels right, do it, otherwise having similar functions for each site with differences in some lines for getting the right url parameters is fine.

As for performance, I'm not sure we'll be able to see the difference on desktop/laptops, but it might matter on phones. It might be too much work to actually benchmark the thing so don't worry too much about it, but anyway the argument of using URLSearchParams objects to avoid potential optional search params or to still work when they've been re-ordered might be enough to avoid the regex.
